hoomd/hpmc/integrate_nec.py

```diff
@@ -0,0 +1,300 @@
+
+from hoomd.hpmc.integrate import HPMCIntegrator
+from hoomd.data.parameterdicts import TypeParameterDict, ParameterDict
+from hoomd.data.typeparam import TypeParameter
+from hoomd.data.typeconverter import OnlyTypes
+
+from hoomd.logging import log
+
+
+class HPMCNECIntegrator(HPMCIntegrator):
+    """HPMC Chain Integrator Meta Class.
+
+    Insert Doc-string here.
+    """
+    _cpp_cls = None
+
+    def __init__(self,
+                 d=0.1,
+                 a=0.1,
+                 chain_probability=0.5,
+                 chain_time=0.5,
+                 update_fraction=0.5,
+                 nselect=1):
+        # initialize base class
+        super().__init__(d, a, 0.5, nselect)
+
+        # Set base parameter dict for hpmc chain integrators
+        param_dict = ParameterDict(
+            chain_probability=OnlyTypes(
+                float, postprocess=self._process_chain_probability),
+            chain_time=OnlyTypes(float, postprocess=self._process_chain_time),
+            update_fraction=OnlyTypes(
+                float, postprocess=self._process_update_fraction))
+        self._param_dict.update(param_dict)
+        self.chain_probability = chain_probability
+        self.chain_time = chain_time
+        self.update_fraction = update_fraction
+
+    def _process_chain_probability(self, value):
+        if 0.0 < value <= 1.0:
+            return value
+        else:
+            raise ValueError(
+                "chain_probability has to be between 0 and 1 (got {}).".format(
+                    value))
+
+    def _process_chain_time(self, value):
+        if 0.0 <= value:
+            return value
+        else:
+            raise ValueError(
+                "chain_time has to be positive (got {}).".format(value))
+
+    def _process_update_fraction(self, value):
+        if 0.0 < value <= 1.0:
+            return value
+        else:
+            raise ValueError(
+                "update_fraction has to be between 0 and 1. (got {})".format(
+                    value))
+
+    def _attach(self):
+        super()._attach()
+
+    @property
+    def nec_counters(self):
+        """Trial move counters.
+
+        The counter object has the following attributes:
+
+        * ``chain_start_count``:        `int` Number of chains
+        * ``chain_at_collision_count``: `int` Number of collisions
+        * ``chain_no_collision_count``: `int` Number of chain events that are
+          no collision (i.e. no collision partner found or end of chain)
+        * ``distance_queries``:         `int` Number of sweep distances
+          calculated
+        * ``overlap_errors``:           `int` Number of errors during sweep
+          calculations
+
+        Note:
+            The counts are reset to 0 at the start of each
+            `hoomd.Simulation.run`.
+        """
+        if self._attached:
+            return self._cpp_obj.getNECCounters(1)
+        else:
+            return None
+
+    @log
```

Comment:
See the new `requires_run` optional argument to `hoomd.logging.log`. Change the other logged properties that need it too.